### PR TITLE
Correctly treat empty encoded string as invalid

### PR DESCRIPTION
**Pull request by jchia**
Compressing empty input wil now yield something. Decompressing empty
input is treated as an error.

### CODE REVIEW

**Comment by jchia** [comment]
#13

File: src/Data/Conduit/Lzma.hs [hunk 1]
```diff
@@ -107,14 +107,12 @@ compress
...
   mval <- await
-  case mval of
-    Just input -> do
-      (streamKey, (streamPtr, _)) <- lift $ allocate
-        (initStream "lzma_easy_encoder" (easyEncoder level))
-        (\ (streamPtr, buffer) -> c'lzma_end streamPtr >> free streamPtr >> free buffer)
-      codeEnum streamKey streamPtr input
-
-    Nothing -> return ()
+  process $ maybe B.empty id mval
```
Comment (NathanHowell) [inline src/Data/Conduit/Lzma.hs]: How about using `fromMaybe :: a -> Maybe a -> a` instead of `maybe`?

Comment (jchia) [inline src/Data/Conduit/Lzma.hs]: Done

**Comment by NathanHowell** [comment]
LGTM 👍  thanks!

**Pull request merged**